contracts/colonyNetwork/ColonyNetworkDeployer.sol

```diff
@@ -34,7 +34,9 @@ contract ColonyNetworkDeployer is ColonyNetworkStorage {
     metaColony = createColony(_tokenAddress, currentColonyVersion, "", "");
 
     // Add the special mining skill, parent is the root domain
-    reputationMiningSkillId = IColonyNetwork(address(this)).addSkill(skillCount - 1);
+    if (isMiningChain()) {
```

Comment:
What are the scenario(s) that you're worried about here? If the mining chain changes, I think a new mining skill is required (otherwise you are earning reputation in the 'wrong' metacolony). The scenario that I see unaccounted for here is if mining is moved to a chain that already has a metacolony, but am wondering if there are others.

There is an implication here that the (main) metacolony and mining are on the same chain, but that's an assumption we've certainly made for a long time.